src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig

```diff
@@ -274,7 +272,11 @@ file that was distributed with this source code.
                 return true;
             });
 
-            {% if sonata_admin.field_description and sonata_admin.field_description.hasAssociationAdmin %}
+            // Automatically select the created record after closing the popup window
+            {% set create_url_exists = sonata_admin.field_description and sonata_admin.field_description.hasAssociationAdmin and btn_add 
```

Comment:
```suggestion
            {% if sonata_admin.field_description and sonata_admin.field_description.hasAssociationAdmin 
            and btn_add 
            and sonata_admin.field_description.associationadmin.hasRoute('create')
            and sonata_admin.field_description.associationadmin.hasAccess('create')
```

Reply:
No need for an extra variable, please use comments if you feel this is unclear.

Reply:
changes applied